### PR TITLE
Upgrade lua to 5.4.2 to fix CVE-2020-15945, bump dependent packages

### PR DESCRIPTION
**Pull request by thcrain-msft**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

The patch for CVE-2020-15945 is not easily ported to our old lua version, so we upgrade to the newest version. We are taking on the risk of a minor version upgrade due to (a) the low number of lua-dependent packages and (b) the usage of Lua backward compatibility flags when building.

###### Change Log  

- Upgrade lua to 5.4.2 to fix CVE-2020-15945
- Bump release of packages depending on lua: haproxy, httpd, nmap, rpm

###### Does this affect the toolchain?  

**YES**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2020-15945

###### Test Methodology

- Local toolchain build + build of affected packages currently ongoing

### CODE REVIEW

**Comment by thcrain-msft** [comment]
Ignoring linter output for release-bumped packages- trying to touch as little as possible about these, other than rebuilding against new version of lua.

**Review by MateuszMalisz — APPROVED** [review]

File: SPECS/lua/lua-5.4.2-shared_library-1.patch [hunk 1]
```diff
@@ -0,0 +1,78 @@
+Submitted By:            Douglas R. Reno <renodr at linuxfromscratch dot org>
+Date:                    2020-06-30
+Initial Package Version: 5.4.0
+Upstream Status:         Rejected
```
Comment (jslobodzian) [inline SPECS/lua/lua-5.4.2-shared_library-1.patch:4]: What should we make of this rejected upstream status?

**Comment by jslobodzian** [comment]
> Requires:       lua

I think that we want to add a specific version here (and the other spec files).  Something like "requires lua 5.4.2."  (Although I think we should have a conversation about this)

---
Refers to: SPECS/rpm/rpm.spec:21 in 7768b33. [](commit_id = 7768b3339cb25711f814e4bcdec1982a54672977, deletion_comment = False)

File: SPECS/lua/lua-5.4.2-shared_library-1.patch (hunk 1 above)
Comment (thcrain-msft) [reply SPECS/lua/lua-5.4.2-shared_library-1.patch:4]: As far as I could tell, upstream does not ship a shared library for Lua and has no desire to at this time.  I'd rather not remove this patch for backwards-compatibility purposes.

**Comment by thcrain-msft** [comment]
Closing- will be handled by #1645

**Pull request closed without merging**